modules/ebi-metagenomics/eggnogmapper/main.nf

```diff
@@ -9,38 +9,40 @@ process EGGNOGMAPPER {
 
     input:
     tuple val(meta), path(fasta)
-    path(eggnog_db)
+    tuple val(meta), path(annotation_hit_table)
     path(eggnog_data_dir)
+    path(eggnog_db)
     path(eggnog_diamond_db)
 
     output:
-    tuple val(meta), path("*.emapper.hits"), emit: csv
-    path "versions.yml",                     emit: versions
+    tuple val(meta), path("*.emapper.hits"), emit: hits, optional: true
+    tuple val(meta), path("*annotations*") , emit: annotations, optional: true
+    tuple val(meta), path("*orthologs*")   , emit: orthologs, optional: true
+    path "versions.yml"                    , emit: versions
 
     when:
     task.ext.when == null || task.ext.when
 
     script:
     def args = task.ext.args ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
-    def is_compressed = fasta.name.endsWith(".gz")
-    def fasta_name = fasta.name.replace(".gz", "")
```

Comment:
Is it difficult to maintain the compressed fasta support with the new functionality?